internal/cmd/agent/trigger/watcher.go

```diff
@@ -125,10 +131,36 @@ func (t *Trigger) OnChange(key string, defaultNamespace string, trigger func(),
 	return nil
 }
 
-func (t *Trigger) call(gvk schema.GroupVersionKind, key objectset.ObjectKey) {
+func (t *Trigger) call(gvk schema.GroupVersionKind, obj metav1.Object, deleted bool) {
```

Comment:
How feasible would it be to cover this with tests? I for one would then feel more comfortable that this reduces triggering without introducing any important, undesired side-effects.
eg.:
* deleted object → triggers called
* added object → triggers not called
* modified object with no generation field → triggers called
* modified object with generation field set to 0 → triggers not called
* modified object with generation field updated → triggers called (possibly repeated a couple of times to validate that subsequent generation updates have the desired effect)

Reply:
Tests added, thanks for the suggestion. It made me find that there was a corner case in which a trigger would happen but is not desired. Please take another look!

Reply:
One of the tests is flaky, I'm looking into it

Reply:
Fixed in ad49a1638a0aa564300d3cd290e0dbf97da64e3c